### PR TITLE
executor: check schema when build table reader executor

### PR DESCRIPTION
**Pull request by yibin87**
### What problem does this PR solve?

Issue Number: ref #50358

Problem Summary:
Described in #50358
### What changed and how does it work?
Quick fix for the issue in executor. Had tried to fix it in planner phase, but failed with so many test failures: https://github.com/pingcap/tidb/pull/50410
May try to fix it in planner phase later.

### Check List

Tests 

- [x] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
Try to run cross join queries with high concurrency, and check no empty result sets are found
- [ ] No need to test
  > - [ ] I checked and no code files have been changed.
  > 

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: pkg/executor/builder.go [hunk 1]
```diff
@@ -3416,6 +3416,10 @@ func (b *executorBuilder) buildTableReader(v *plannercore.PhysicalTableReader) e
 			failpoint.Return(nil)
 		}
 	})
+	// https://github.com/pingcap/tidb/issues/50358
```
Comment (elsa0520) [inline pkg/executor/builder.go:3419]: Please add the test ~

Comment (yibin87) [reply pkg/executor/builder.go:3419]: ok

**Review by elsa0520 — CHANGES_REQUESTED** [review]
This is not a good changed for this issue.
We should keep the tablereader schema correctly in planner.

**Comment by yibin87** [comment]
> This is not a good changed for this issue. We should keep the tablereader schema correctly in planner.

Yeah, that's the reason why this is not a PR that closes the issue but ref it. This PR aims to pass some regression tests. I tried  to fix the issue in planner phase in #50410 and recorded some other tries, while introducing many test failures. Thus I submit this PR to pass regression tests.

**Comment by elsa0520** [comment]
> > This is not a good changed for this issue. We should keep the tablereader schema correctly in planner.
> 
> Yeah, that's the reason why this is not a PR that closes the issue but ref it. This PR aims to pass some regression tests. I tried to fix the issue in planner phase in #50410 and recorded some other tries, while introducing many test failures. Thus I submit this PR to pass regression tests.

OK, for quickly pass the test I can approve this PR. But we should directly fix this issue by following planner PR later.

**Review by elsa0520 — APPROVED** [review]
Later to fix it correctly.

**Review by windtalker — APPROVED** [review]
LGTM

**Pull request merged**